Model/Api/ShippingMethods.php

```diff
@@ -743,6 +755,27 @@ public function getShippingOptions($quote, $addressData)
         return $shippingMethods;
     }
 
+    /**
+     * @param $quote
+     * @return bool
+     */
+    protected function doesDiscountApplyToShipping($quote){
+        if ($quote->getAppliedRuleIds()) {
```

Comment:
Could I propose to define a variable for applied rules before `if` so that it can be used inside `foreach` later? It's only my opinion, but I believe it would help keep code readability and clearance.

Reply:
Just removing the `if` condition would do.

Reply:
> Just removing the `if` condition would do.

Since the `getAppliedRuleIds` method could return either string or `null`, there's the sense to check the value it returns, so it would be good to use `if` here to avoid an extra cast its value to the string.